pod/pod.go

```diff
@@ -1,15 +1,59 @@
 package main
 
 import (
+	"log"
 	"os"
 	"os/signal"
 	"syscall"
 )
 
+func reapChildProcesses() {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGCHLD)
+
+	for {
+		// Block until SIGCHLD signal is received.
+		<-c
+		reapChildProcessesHelper()
+	}
+}
+
+func reapChildProcessesHelper() {
+	for {
+		// Pid -1 means to wait for any child process.
+		// With syscall.WNOHANG option set, function will
+		// not block and will exit immediatelly if no child
+		// process has exited.
+		_, err := syscall.Wait4(-1, nil, syscall.WNOHANG, nil)
+
+		switch err {
+		case nil:
+			// One or more child process still exist,
+			// but have not yet changed state. Try again.
+		case syscall.EINTR:
+			// Got interrupted. Shouldn't happen with WNOHANG option,
+			// but it is better to handle it anyway and try again.
+		case syscall.ECHILD:
+			// No more child processes to reap. We can return and wait
+			// for another SIGCHLD signal.
+			return
+		default:
+			// Some other unexpected error. Return and wait for
+			// another SIGCHLD signal.
+			log.Println(err)
+			return
+		}
+	}
+}
+
 func main() {
 	c := make(chan os.Signal, 1)
 	signal.Notify(c, os.Interrupt, os.Kill, syscall.SIGTERM)
 
-	// Block until a signal is received.
+	if os.Getpid() == 1 {
```

Comment:
Add a log that we are starting the reaper.

Reply:
Done